.github/workflows/publish.yml

```diff
@@ -44,3 +44,27 @@ jobs:
           push: true
           tags: ${{ steps.docker_metadata.outputs.tags }}
           labels: ${{ steps.docker_metadata.outputs.labels }}
+
+  deploy-k8s-staging:
+    runs-on: ubuntu-22.04
+    needs:
+      - publish-docker-images
+    env:
+      KUBECTL_VERSION: v1.21.4
+      DATAVERSE_PORTAL_IMAGE: ghcr.io/okp4/dataverse-portal
+    steps:
+      - name: Set up kubectl
+        run: |
+          curl -fsSL -o ./kubectl https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl
+          chmod +x ./kubectl
+          sudo mv ./kubectl /usr/bin/kubectl
+
+      - name: Set up kubernetes config
+        run: |
+          echo ${{ secrets.KUBECONFIG }} > config_enc
+          cat config_enc | base64 -d > config
+          rm config_enc
+
+      - name: Deploy staging
+        run: |
+          kubectl set image deployment/dataverse-portal dataverse-portal=${DATAVERSE_PORTAL_IMAGE}:nightly-${{ github.sha }} -n staging
```

Comment:
We just need to update a metadata in order to trigger the rolling update:
```sh
kubectl patch deploy deploy dataverse-portal -p '{metadata:{"labels": {"commit": "${{ github.sha }}"}}}'
```

Reply:
Ok I just understood what you meant :open_mouth: .
Very smart thanks no need to change the tag  :100: !

Reply:
Yes there is 😆